app/lib/three_scale/spam_protection/integration/controller.rb

```diff
@@ -20,8 +20,9 @@ def spam_check(object, session_store: ThreeScale::SpamProtection::SessionStore)
         when :none
           true
         when :auto
-          if object.spam?
-            session_store.new(request.session).mark_possible_spam
+          stored_session = session_store.new(request.session)
+          if object.spam? && stored_session.marked_as_possible_spam?
+            stored_session.mark_possible_spam
```

Comment:
This is a fix for the bug.

If the credentials were correct it would mark it as spam because we are passing a dummy (empty) model to it when the user is not logged in.